src/ng/directive/input.js

```diff
@@ -1290,6 +1290,9 @@ function createDateParser(regexp, mapping) {
           }
         });
         return new Date(map.yyyy, map.MM - 1, map.dd, map.HH, map.mm, map.ss || 0, map.sss * 1000 || 0);
+      } else {
+        // If we fail to parse the string with the ISO regex, it might be because it's the result of Date.toString()
+        return Date.parse(iso);
```

Comment:
This can lead to unexpected results. E.g. `Date.parse('0')` is the same as `Date.parse('2000')` which is the same as `2000-01-01T00:00:00` in local time - which is probably not the intended value :smiley:

If this is only to facilitate our new `toString()` stringification?